ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/forkchoice/ForkChoice.java

```diff
@@ -405,6 +390,31 @@ private void notifyForkChoiceUpdatedAndOptimisticSyncingChanged() {
     }
   }
 
+  private void onExecutionPayloadResult(
+      final Bytes32 blockRoot, final PayloadStatus result, final UInt64 latestFinalizedBlockSlot) {
+    onForkChoiceThread(
+            () -> {
+              if (!result.hasValidStatus()) {
+                return;
+              }
```

Comment:
I don't think we should be ignoring non-valid statuses.  In fact I think if we get an invalid response we need to immediately run fork choice again and update the chain head because we've potentially just re-orged over to a branch that has turned out to be invalid and want to get back off of it.

I recently fixed this in our existing code but I think it becomes more important now that the invalid response may come from fork choice updates.

Reply:
ouch, bad rebase.